### DISCUSSION TITLE
PushNotifications still supported?

### DISCUSSION THREAD

**Discussion by AffluentOwl** [microsoft/WindowsAppSDK#1960] [Q&A] [answered]
Is it currently possible to test out PushNotifications?

The [quickstart guide](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/notifications/push/push-quickstart) says a private preview form needs to be filled out, but the link appears to be broken, https://aka.ms/windowsappsdk/push-private-preview .

Additionally, can you confirm that this feature should be available in 1.0.0-experimental1? Or was it only available in a previous experimental version? Perhaps I've messed something up when attempting to uninstall 1.0.0 and move to 1.0.0-experimental. I am attempting to use the Microsoft.Windows.PushNotifications interface in Visual Studio 2022, with t he 1.0.0-experimtanl C# template for a MSIX project.

`Error	CS0234	The type or namespace name 'Windows' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?)`

Curiously, other Windows App SDK dependencies work:

```
           Microsoft.UI.Windowing.AppWindow a;
            Microsoft.ApplicationModel.Resources.ResourceManager m;

```

While these fail to compile:

```
            Microsoft.Windows.AppLifecycle l;
            Microsoft.ApplicationModel.ApplicationLifetime al;
            Microsoft.Windows.PushNotifications p;
```

I also see in my packages only the following DLL's loaded, but my understanding is that none of these are the DLL which contains PushNotifications support.

```
C:\Users\xxx\.nuget\packages\microsoft.windowsappsdk.foundation\1.0.0-experimental1\lib\net5.0-windows10.0.17763.0\Microsoft.ApplicationModel.Resources.Projection.dll
C:\Users\xxx\.nuget\packages\microsoft.windowsappsdk.interactiveexperiences\1.0.0-experimental1\lib\net5.0-windows10.0.17763.0\Microsoft.InteractiveExperiences.Projection.dll
C:\Users\xxx\.nuget\packages\microsoft.windowsappsdk.winui\1.0.0-experimental1\lib\net5.0-windows10.0.18362.0\Microsoft.WinUI.dll
```

I also notice the message in the NuGet package manager in Visual Studio which says in italics `Microsoft.WindowsAppSDK.Runtime - not available in this source`. Whereas the other 3 packages without such a message correspond to the 3 DLL's I was able to get.

**Comment by nicuparente** [comment, marked as answer]
Hi @AffluentOwl - Are you trying to use C# for Push Notifications 1.0.0-experimental1? Only C++ is supported on the 1.0.0-exp1 release, C# will be supported on the next experimental release. 

Thanks for pointing out the broken link, we will be fixing it.

**Reply by jacksonb-opentext** [reply]
What is the estimate on getting the push preview link fixed? I have been trying it for over a week and it hasn't been working.

**Reply by nicuparente** [reply]
I appreciate the patience. We are working towards removing the Private Preview requirement very soon and we will be updating the docs to remove the Private Preview sign up requirements. I'll update this thread once we have completed it.

**Comment by AffluentOwl** [comment]
Thanks for the quick response, that was indeed the issue. I felt I read the documentation quite extensively, but didn't see anything mentioning that 1.0.0-experimental1 was only available for C++, and it felt quite unexpected to me since the C# templates are included under the experimental channel.  Or was it just the PushNotifications feature that was not supported in C#? Sounds like it will be a non-issue near future for others in the when the next release comes out.

Is there any word out on the street for when the next experimental build will be out? Or what's the best way for me to get informed when a target date becomes available?

Is it possible for me to build my own WindowsAppSDK build from head with C# now? If so, are there any instructions you could point me towards?

PushNotifications are a huge win for me, since it allows me to radically redesign and simplify my app/server architecture, letting me go with a serverless design, and avoiding long running TCP connections. So I'm excited to try it out as soon as possible.

**Reply by nicuparente** [reply]
Thanks for sharing how push will be able to help your scenario! We are really excited for you to be able to try it out soon!

"Is there any word out on the street for when the next experimental build will be out? Or what's the best way for me to get informed when a target date becomes available?"
>> No, official dates on the experimental release yet and aside from the docs, this Github Repo discussion section has the most up to date announcements on our releases. 

"Is it possible for me to build my own WindowsAppSDK build from head with C# now? If so, are there any instructions you could point me towards?"
>> No, this is currently not possible.
